plugins/init/Drupal8.civi-setup.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * @file
+ *
+ * Determine default settings for Drupal 8.
+ */
+
+if (!defined('CIVI_SETUP')) {
+  exit("Installation plugins must only be loaded by the installer.\n");
+}
+
+\Civi\Setup::dispatcher()
+  ->addListener('civi.setup.checkAuthorized', function (\Civi\Setup\Event\CheckAuthorizedEvent $e) {
+    $model = $e->getModel();
+    if ($model->cms !== 'Drupal8' || !function_exists('user_access')) {
+      return;
+    }
+
+    \Civi\Setup::log()->info(sprintf('[%s] Handle %s', basename(__FILE__), 'checkAuthorized'));
+    $e->setAuthorized(user_access('administer modules'));
+  });
+
+  \Civi\Setup::dispatcher()
+    ->addListener('civi.setup.init', function (\Civi\Setup\Event\InitEvent $e) {
+      $model = $e->getModel();
+      $cmsPath = \Drupal::root();
```

Comment:
This line should be moved down a few lines. The `civi.setup.init` event will fire in all contexts -- if you tried to run `\Drupal::root()` on a D7/Backdrop/WordPress/Joomla site, it would cause a hard crash.

Reply:
Ahh right thanks for noticing that. Done